assets/src/edit-story/components/canvas/page.js

```diff
@@ -44,32 +45,59 @@ function Page() {
 		actions: { clearSelection, selectElementById, toggleElementIdInSelection },
 	} = useStory();
 	const {
-		actions: { setBackgroundClickHandler },
+		state: { isEditing, editingElement },
+		actions: { setBackgroundClickHandler, setEditingElement },
 	} = useCanvas();
-	const handleSelectElement = useCallback( ( id ) => ( evt ) => {
+	const handleSelectElement = useCallback( ( id, evt ) => {
 		if ( evt.metaKey ) {
 			toggleElementIdInSelection( id );
 		} else {
 			selectElementById( id );
 		}
 		evt.stopPropagation();
 	}, [ toggleElementIdInSelection, selectElementById ] );
-	const selectionProps = hasSelection ? getUnionSelection( selectedElements, 2 ) : {};
+	const handleEnterEditMode = ( id ) => {
+		// Only edited element is selected.
+		selectElementById( id );
+		setEditingElement( id );
+	};
+	const getClickHandler = useDoubleClick( handleSelectElement, handleEnterEditMode );
+	const selectionProps = hasSelection ? getUnionSelection( selectedElements ) : {};
 	useEffect( () => {
 		setBackgroundClickHandler( () => clearSelection() );
 	}, [ setBackgroundClickHandler, clearSelection ] );
 	return (
 		<Background>
 			{ currentPage && currentPage.elements.map( ( { type, id, ...rest } ) => {
-				const comp = getComponentForType( type );
-				const Comp = comp; // why u do dis, eslint?
+				const { hasEditMode, Display, Edit } = getDefinitionForType( type );
+
+				// Are we editing this element, display this as Edit component.
+				if ( editingElement === id ) {
+					return (
+						<Element key={ id }>
+							<Edit { ...rest } />
+						</Element>
+					);
+				}
+
+				// Are we editing some other element, display this as passive Display.
```

Comment:
Could you clarify what's this mode does exactly?

Reply:
`isPassive` here on the individual element just refers to [the element being faded a bit](https://github.com/ampproject/amp-wp/pull/3804/files#diff-3688303d740704ce521b8f44dbb09923R37).